src/cargo/core/profiles.rs

```diff
@@ -858,36 +878,65 @@ impl UnitFor {
         }
     }
 
-    /// Returns `true` if this unit is for a custom build script or one of its
-    /// dependencies.
-    pub fn is_build(self) -> bool {
-        self.build
+    /// Returns a new copy updating it for a build dependency.
+    ///
+    /// This is part of the machinery responsible for handling feature
+    /// decoupling for build dependencies in the new feature resolver.
+    pub fn with_build_dep(mut self, build_dep: bool) -> UnitFor {
```

Comment:
As a minor nuance (and this was already present, just something I'm realizing now) the terminology of `with_*` doesn't quite fit here in my mind since I'd expect a function of that name to unconditionally configure `build_dep` with the argument provided, but here it's more of a unioning function. I'm not sure of a better name though!

Reply:
Yea, I struggled a bit with the naming convention and wasn't happy with it.  But I'm unable to think of anything particularly better.  If you think of something, let me know!

I don't know if there is a common naming convention for this idiom (creating a copy and modifying a field in one step)?  Maybe that's a sign it may be better to be explicit about creating a copy, and then call `set_build_dep`?

Reply:
Eh it's a minor thing, I wouldn't stress out much over it. I definitely know of no naming convention for this, so it may be good to just highlight it in the documentation and move on. (the current call-sites all "look right" which is what matters)